src/suites/cts/copyImageBitmapToTexture.spec.ts

```diff
@@ -0,0 +1,83 @@
+export const description = `
+copy imageBitmap To texture tests.
+`;
+
+import { TestGroup } from '../../framework/index.js';
+
+import { GPUTest } from './gpu_test.js';
+
+export const g = new TestGroup(GPUTest);
+
+function calculateRowPitchForImageBitmap(width: number): number {
+  if (width % 256 === 0) {
+    return 4 * width;
+  }
+  return 4 * (width + 256 - (width % 256));
+}
+
+g.test('from image element', async t => {
+  const img = document.createElement('img');
+  img.src = '../../assets/img/test.png';
+  await img.decode();
+
+  const imageBitmap = await createImageBitmap(img, 0, 0, img.width, img.height);
+  const dst = t.device.createTexture({
+    size: {
+      width: img.width,
+      height: img.height,
+      depth: 1,
+    },
+    format: 'rgba8uint',
+    usage: GPUTextureUsage.COPY_DST | GPUTextureUsage.COPY_SRC,
+  });
+
+  // TODO(shaobo.yan@intel.com) : remove as any after @webgpu/types updating
+  //(t.device.defaultQueue as any).copyImageBitmapToTexture(
+  t.device.defaultQueue.copyImageBitmapToTexture(
+    {
+      imageBitmap,
+      origin: {
+        x: 0,
+        y: 0,
+      },
+    },
+    {
+      texture: dst,
+    },
+    {
+      width: imageBitmap.width,
+      height: imageBitmap.height,
+      depth: 1,
+    }
+  );
+
+  const imageCanvas = document.createElement('canvas');
+  imageCanvas.width = imageBitmap.width;
+  imageCanvas.height = imageBitmap.height;
+
+  const imageCanvasContext = imageCanvas.getContext('2d');
+  t.expect(imageCanvasContext != null);
+  if (!imageCanvasContext) {
+    return;
+  }
+
+  imageCanvasContext.drawImage(imageBitmap, 0, 0, imageBitmap.width, imageBitmap.height);
+  const data = imageCanvasContext.getImageData(0, 0, imageBitmap.width, imageBitmap.height).data;
+
+  const rowPitchValue = calculateRowPitchForImageBitmap(imageBitmap.width);
+  const testBuffer = t.device.createBuffer({
+    size: rowPitchValue * imageBitmap.height,
+    usage: GPUBufferUsage.COPY_SRC | GPUBufferUsage.COPY_DST,
+  });
+
+  const encoder = t.device.createCommandEncoder();
+
+  encoder.copyTextureToBuffer(
+    { texture: dst, mipLevel: 0, origin: { x: 0, y: 0, z: 0 } },
+    { buffer: testBuffer, rowPitch: rowPitchValue, imageHeight: imageBitmap.height },
```

Comment:
Do we allow an imageHeight of 0? There's no reason to actually provide an imageHeight here since it's not an array or 3d texture.

Opened https://github.com/gpuweb/gpuweb/pull/517

Reply:
Agree with you. I feel a little bit confusing when I write this copy.

Reply:
If imageBitmap:0 is allowed in chrome, let's use it just to simplify.

Reply:
Ack.

Reply:
Done